### PR TITLE
macs2: init at 2.2.6

### PR DESCRIPTION
**Pull request by GregorySchwartz**
###### Motivation for this change

`MACS2` is a commonly used tool for peak finding in ChIP-seq data.

###### Things done

`MACS2` is a python application so I did the (what I interpret as) basic procedure to add the package. 

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions (Arch Linux)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/science/biology/macs2/default.nix [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    doCheck = false;
+
+    meta = with lib; {
+      description = "Model-based Analysis for ChIP-Seq";
+      license = licenses.bsd3;
+      maintainers = with maintainers; [ gschwartz ];
+      platforms = platforms.linux;
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/macs2/default.nix:17]:
```suggestion
  };
}
```

File: pkgs/applications/science/biology/macs2/default.nix [hunk 2]
```diff
@@ -0,0 +1,17 @@
+{ pkgs, lib, buildPythonPackage ? pkgs.python3.pkgs.buildPythonPackage, fetchurl, numpy ? pkgs.python3.pkgs.numpy}:
+
+buildPythonPackage rec {
+    pname = "macs2";
+    version = "2.2.6";
+    src = fetchurl { url = "https://files.pythonhosted.org/packages/21/0f/972b44c84d85e37d816beae88aa5ddad606bd757630d77dc2f558900a6ce/MACS2-2.2.6.tar.gz"; sha256 = "ca9e43f8cf15312ace9afa0a9bf67e8072227376235b7b0f18dd2dcac17dea42";};
+    propagatedBuildInputs = [ numpy ];
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/macs2/default.nix:7]:
```suggestion
    version = "2.2.6";

    src = fetchurl {
      url = https://files.pythonhosted.org/packages/21/0f/972b44c84d85e37d816beae88aa5ddad606bd757630d77dc2f558900a6ce/MACS2-2.2.6.tar.gz";
      sha256 = "ca9e43f8cf15312ace9afa0a9bf67e8072227376235b7b0f18dd2dcac17dea42";
    };

    propagatedBuildInputs = with python3.pkgs; [ numpy ];
```

Please use fetchpypi.

File: pkgs/applications/science/biology/macs2/default.nix [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{ pkgs, lib, buildPythonPackage ? pkgs.python3.pkgs.buildPythonPackage, fetchurl, numpy ? pkgs.python3.pkgs.numpy}:
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/macs2/default.nix:1]:
```suggestion
{ lib, python3, fetchurl }:
```

File: pkgs/applications/science/biology/macs2/default.nix [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+buildPythonPackage rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/macs2/default.nix:3]:
```suggestion
python3.pkgs.buildPythonPackage rec {
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 118610` at 5f4f33ae run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>macs2</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/applications/science/biology/macs2/default.nix:11:0:
```
   |
11 |       description = "Model-based Analysis for ChIP-Seq";
   | ^
```
</li>
  </ul>
</details>

**Comment by GregorySchwartz** [comment]
If this is an application, does `pythonImportsCheck` make sense at all?

File: pkgs/applications/science/biology/macs2/default.nix [hunk 5]
```diff
@@ -0,0 +1,21 @@
+{ lib, python3, fetchurl }:
+
+python3.pkgs.buildPythonPackage rec {
+    pname = "macs2";
+    version = "2.2.6";
+
+    src = fetchurl {
+      url = https://files.pythonhosted.org/packages/21/0f/972b44c84d85e37d816beae88aa5ddad606bd757630d77dc2f558900a6ce/MACS2-2.2.6.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/macs2/default.nix:8]:
```suggestion
      url = "https://files.pythonhosted.org/packages/21/0f/972b44c84d85e37d816beae88aa5ddad606bd757630d77dc2f558900a6ce/MACS2-2.2.6.tar.gz";
```

**Comment by SuperSandro2000** [comment]
> If this is an application, does `pythonImportsCheck` make sense at all?

If it can be imported yes. If it can't be its alright, too.

File: pkgs/applications/science/biology/macs2/default.nix [hunk 6]
```diff
@@ -0,0 +1,21 @@
+{ lib, python3, fetchurl }:
+
+python3.pkgs.buildPythonPackage rec {
+    pname = "macs2";
+    version = "2.2.6";
+
+    src = fetchurl {
+      url = "https://files.pythonhosted.org/packages/21/0f/972b44c84d85e37d816beae88aa5ddad606bd757630d77dc2f558900a6ce/MACS2-2.2.6.tar.gz";
+      sha256 = "ca9e43f8cf15312ace9afa0a9bf67e8072227376235b7b0f18dd2dcac17dea42";
+    };
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/macs2/default.nix:10]: Please convert this to use fetchPypi.

Comment (GregorySchwartz) [reply pkgs/applications/science/biology/macs2/default.nix]: Done.

**Comment by GregorySchwartz** [comment]
> > If this is an application, does `pythonImportsCheck` make sense at all?
> 
> If it can be imported yes. If it can't be its alright, too.

Used for MACS2.

File: pkgs/applications/science/biology/MACS2/default.nix [hunk 7]
```diff
@@ -0,0 +1,23 @@
+{ lib, python3, fetchurl }:
+
+python3.pkgs.buildPythonPackage rec {
+    pname = "MACS2";
+    version = "2.2.7.1";
+
+    src = python3.pkgs.fetchPypi {
+      inherit version;
+      inherit pname;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/MACS2/default.nix:9]:
```suggestion
      inherit pname version;
```

File: pkgs/applications/science/biology/MACS2/default.nix [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+      sha256 = "1rcxj943kgzs746f5jrb72x1cp4v50rk3qmad0m99a02vndscb5d";
+    };
+
+    propagatedBuildInputs = with python3.pkgs; [ numpy ];
+    doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/MACS2/default.nix:14]: Please add a comment why you disabled tests.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please squash all package related commits together so that we have 2 commits in the end: one for the package and one for adding you as a maintainer.

File: pkgs/applications/science/biology/MACS2/default.nix (hunk 8 above)
Comment (GregorySchwartz) [reply pkgs/applications/science/biology/MACS2/default.nix:16]: Done

File: pkgs/applications/science/biology/MACS2/default.nix [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    # function (ImportError: Failed to import test module: diffpeak_cmd)
+    doCheck = false;
+    pythonImportsCheck = [ "MACS2" ];
+
+    meta = with lib; {
+      description = "Model-based Analysis for ChIP-Seq";
+      license = licenses.bsd3;
+      maintainers = with maintainers; [ gschwartz ];
+      platforms = platforms.linux;
+  };
```
Comment (SuperSandro2000) [inline pkgs/applications/science/biology/MACS2/default.nix:24]: Please intent with 2 spaces per level like the editor config suggests.

File: pkgs/applications/science/biology/MACS2/default.nix [hunk 10]
```diff
@@ -0,0 +1,25 @@
+{ lib, python3, fetchurl }:
+
+python3.pkgs.buildPythonPackage rec {
```
Comment (dotlambda) [inline pkgs/applications/science/biology/MACS2/default.nix:3]:
```suggestion
python3.pkgs.buildPythonApplication rec {
```

**Pull request merged**